### ISSUE TITLE
windows build error

### ISSUE THREAD

**Issue by vadiks2003** [tesseract-ocr/tesseract#4166]
### Current Behavior

i downloaded the recent source files. followed the guide where i downloaded sw and ran sw build inside tesseract folder. ran both sw build and cmake -B buildfolder and tried to run cmake with gcc/g++ yet i get this error

Exception in file D:/dev/cppan2/client2/src/sw/builder/command.cpp:840, function execute1: When executing: [pub.egorpugin.primitives.command-0.3.1]/src/command.cpp
C:/Users/vadik/.sw/storage/pkg/9d/a6/8d0d/src/sdir/src/command/src/command.cpp(142): error C2039: 'contains': is not a member of 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>'
C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include\xstring(4871): note: see declaration of 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>'
C:/Users/vadik/.sw/storage/pkg/9d/a6/8d0d/src/sdir/src/command/src/command.cpp(147): note: see reference to function template instantiation 'auto primitives::Command::setProgram::<lambda_1>::()::<lambda_1>::operator ()<const char(&)[6]>(_T1) const' being compiled
        with
        [
            _T1=const char (&)[6]
        ]
C:/Users/vadik/.sw/storage/pkg/9d/a6/8d0d/src/sdir/src/command/src/command.cpp(147): error C3487: 'bool': all return expressions must deduce to the same type: previously it was 'void'
C:/Users/vadik/.sw/storage/pkg/9d/a6/8d0d/src/sdir/src/command/src/command.cpp(144): error C3487: 'bool': all return expressions must deduce to the same type: previously it was 'void'
C:/Users/vadik/.sw/storage/pkg/9d/a6/8d0d/src/sdir/src/command/src/command.cpp(144): error C2562: 'primitives::Command::setProgram::<lambda_1>::()::<lambda_1>::operator ()': 'void' function returning a value
C:/Users/vadik/.sw/storage/pkg/9d/a6/8d0d/src/sdir/src/command/src/command.cpp(147): note: see declaration of 'primitives::Command::setProgram::<lambda_1>::()::<lambda_1>::operator ()'
C:/Users/vadik/.sw/storage/pkg/9d/a6/8d0d/src/sdir/src/command/src/command.cpp(147): error C2296: '||': illegal, left operand has type 'void'
C:/Users/vadik/.sw/storage/pkg/9d/a6/8d0d/src/sdir/src/command/src/command.cpp(147): error C2297: '||': illegal, right operand has type 'void'
command failed: exit code = 2
Total errors: 1
sw command: C:/TestFolderForPaths/sw.exe -static -platform x64 -compiler msvc-19.29 -d C:/Users/vadik/Downloads/tesseract/tesseracte/build/.sw/cmake -config d,msr,rwdi,r integrate -cmake-deps C:/Users/vadik/Downloads/tesseract/tesseracte/build/.sw/cmake/sw.txt -cmake-file-version 9
CMake Error at C:/Users/vadik/.sw/storage/etc/sw/static/SWConfig.cmake:267 (message):
  sw: non-zero exit code: 1
Call Stack (most recent call first):
  CMakeLists.txt:393 (sw_execute)

### Expected Behavior

its expected to build

### Suggested Fix

change source code

### Operating System

Windows 10

### Compiler

MSVC 2019 14.29.30133

### CPU

Intel(R) Core(TM) i5-1035G1 CPU

### Other Information

windows build is 19045.3693 ver 22H2  Windows 10 Pro N

**Comment by zdenop** [comment]
> C:/Users/vadik/.sw/storage/pkg/9d/a6/8d0d/src/sdir/src/command/src/command.cpp(147): error C2297: '||': illegal, right operand has type 'void'

as you see: this is not tesseract issue but sw - report it there.

**Comment by GitGoliath** [comment]
Hi,

I had the same issue with Leptonica build, did a google search and found this. So for others that face the same issue and stumble on this. Here is a solution...

It was caused by the fact that I ran the CMake without having the SW installed. CMake created file in the build folder with invalid path. After installing the SW client and running again CMake I got the error 267.

I resolved it by erasing all the files in the build folder and relaunch CMake. It worked with a clean folder.
